### PR TITLE
Torque enums

### PR DESCRIPTION
**Pull request by nico-hartmann**
Adds a small chapter about the new enum language feature to the Torque manual

### CODE REVIEW

File: src/docs/torque.md [hunk 1]
```diff
@@ -348,6 +349,81 @@ Call(f, Undefined);
...
+```
+
+Note that `constexpr` is part of the type name and thus goes after the namespace scope.
+
+Torque's enumerations work very well together with the `typeswitch` construct, because the
+values are defined using distinct types:
+
+```torque
+typeswitch(language_mode) {
+  case (LangugageMode::kStrict): {
```
Comment (ripsawridge) [inline src/docs/torque.md]: s/Langugage/Language

File: src/docs/torque.md [hunk 2]
```diff
@@ -348,6 +349,81 @@ Call(f, Undefined);
...
+    // ...
+  }
+  case (LanguageMode::kSloppy): {
```
Comment (ripsawridge) [inline src/docs/torque.md]: Here too

File: src/docs/torque.md [hunk 3]
```diff
@@ -348,6 +349,81 @@ Call(f, Undefined);
...
+type, that is the runtime type used to represent a value of the enum. In this example, this is `TNode<Smi>`,
+since this is what the type `Smi` `generates`. A `constexpr LanguageMode` converts to `LanguageMode`
+in the generated CSA files since no `constexpr` clause is specified on the enum to replace the default name.
+If the `extends` clause is omitted, Torque will generate only the `constexpr` version of the type. The `extern` keyword tells Torque that there is a C++ definition of this enum. Currently, only `extern` enums are supported.
+
+Torque generates a distinct type and constant for each of the enum's entries. Those are defined
+inside a namespace that matches the enum's name. Necessary specializations of `FromConstexpr<>` are
+generated to convert from the entry's `constexpr` types to the enum type. The value generated for an entry in the C++ files is `<enum-constexpr>::<entry-name>` where `<enum-constexpr>` is the `constexpr` name generated for the enum. In the above example, those are `LanguageMode::kStrict` and `LanguageMode::kSloppy`.
+
+To summarize, the above example is desugared into something that looks very similar to this:
```
Comment (ripsawridge) [inline src/docs/torque.md]: I'd consider leaving out the desugared code example. What about just mentioned where the output for enums is written to in gen/torque-generated.

File: src/docs/torque.md (hunk 1 above)
Comment (nico-hartmann) [reply src/docs/torque.md]: Done.

File: src/docs/torque.md (hunk 2 above)
Comment (nico-hartmann) [reply src/docs/torque.md]: Done.

File: src/docs/torque.md (hunk 3 above)
Comment (nico-hartmann) [reply src/docs/torque.md]: There is no distinct file where stuff is written for enums. They appear all over the place where they are used, mostly.

**Pull request merged**